orientation-event/motion/permissions-policy/devicemotion-enabled-on-self-origin-by-permissions-policy.https.html

```diff
@@ -0,0 +1,69 @@
+<!DOCTYPE html>
+<body>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="/permissions-policy/resources/orientation-event.js"></script>
+<script src="/permissions-policy/resources/permissions-policy.js"></script>
+<script src="/common/get-host-info.sub.js"></script>
+<script src="../../resources/orientation-event-helpers.js"></script>
+<script>
+"use strict";
+
+const same_origin_src =
+  "/permissions-policy/resources/permissions-policy-orientation-event.sub.html?eventName=devicemotion";
+const cross_origin_src =
+  get_host_info().HTTPS_REMOTE_ORIGIN + same_origin_src;
+
+const header = 'Permissions-Policy header "accelerometer=self, gyroscope=self"';
+
+promise_test(async t => {
+  const helper = new SensorTestHelper(t, 'devicemotion');
+  await helper.grantSensorsPermissions();
+  await helper.initializeSensors();
```

Comment:
This puts a dependency on create_virtual_sensor :( Are we sure there isn't some other way to test this? 

With other permission policy tests, we usually just have a simple iframe that tries to use the API.

Reply:
> This puts a dependency on create_virtual_sensor :( Are we sure there isn't some other way to test this?

I don't think so. This API does not have any mechanisms to throw an exception when permissions policy checks fail like other APIs do, so we need a way to verify that the checks have failed, in which case we wait for readings _not_ to be delivered. We could try to reach consensus on w3c/deviceorientation#118, but it wouldn't help with the case in which we need to verify that the checks do pass, as without virtual sensors we'd be entirely dependent on the presence of specific hardware for them to work.

Reply:
Ok, yeah... I was trying to think of something too... and the only thing was trying to create the events in JS, which is silly.